src/client/Client.cc

```diff
@@ -6578,7 +6577,7 @@ void Client::tick()
   while (!p.end()) {
     Inode *in = *p;
     ++p;
-    if (!mount_aborted && in->hold_caps_until > now)
+    if (!mount_aborted && in->hold_caps_until > ceph::coarse_mono_clock::now())
```

Comment:
This seems wrong to me, as `now` was determined at a different point than here. @batrick thoughts?

Reply:
@joscollin I guess i should use another variable for comparing with in->hold_caps_until which should be from ceph::coarse_mono_clock::now() coz "now" is still used at many places where and its time from ceph_clock.

Reply:
Check if casting `now` works, as we discussed [1]. It should be fine for now to have minimal change.

[1] https://github.com/ceph/ceph/blob/master/src/client/MetaRequest.cc#L13

Reply:
Just initialize it once and change `last_auto_reconnect` to a monotonic clock?

Reply:
> This seems wrong to me, as `now` was determined at a different point than here. @batrick thoughts?

If use `now` here, this patch won't fix the issue @batrick mentioned in the tracker. We should use the mononic clock here too.

Maybe we should switch the other use cases to mononic clock either.

Reply:
> Just initialize it once and change `last_auto_reconnect` to a monotonic clock?

Changing only last_auto_reconnect is not fixing the issue,op_stamp and client_mount_timepoint also needs change.

Reply:
Maybe you can add a new field in `MetaRequest` class to record the monotonic clock, which will only be used in client side ? Then there is no need to change the `op_stamp` related code in MDS side.

And also the `last_auto_reconnect` and `last_cap_renew` seems having the same issue with `hold_caps_until`.

Reply:
> Maybe you can add a new field in `MetaRequest` class to record the monotonic clock, which will only be used in client side ? Then there is no need to change the `op_stamp` related code in MDS side.
> 
> And also the `last_auto_reconnect` and `last_cap_renew` seems having the same issue with `hold_caps_until`.

Creating another field coarse_stamp in Metarequest.h and using it for comparison.